cub/util_arch.cuh

```diff
@@ -44,12 +44,6 @@ CUB_NAMESPACE_BEGIN
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS    // Do not document
 
-#if ((__CUDACC_VER_MAJOR__ >= 9) || defined(_NVHPC_CUDA) ||            \
-     CUDA_VERSION >= 9000) &&                                                  \
-  !defined(CUB_USE_COOPERATIVE_GROUPS)
-#define CUB_USE_COOPERATIVE_GROUPS
```

Comment:
Leave this defined unconditionally for now and add a note stating that it's deprecated, since users may depend on it. 

Mark this as a breaking change and put a note in the PR description stating that this macro is deprecated. We can fully remove it after a few releases 👍